erpnext/public/js/controllers/transaction.js

```diff
@@ -1760,14 +1760,39 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 	}
 });
 
-erpnext.show_serial_batch_selector = function(frm, d, callback, on_close, show_dialog) {
+erpnext.show_serial_batch_selector = function (frm, d, callback, on_close, show_dialog) {
+	let warehouse, receiving_stock, existing_stock;
+	if (frm.doc.is_return) {
+		receiving_stock = true;
```

Comment:
In case of purchase return, should it be false?

Reply:
You're right, I've handled purchase documents now too with the latest commit.

Although during testing, I noticed that if an item is setup for both batch and serial numbers, the system always defaults to the "Select Batch" prompt. This is problematic for selecting serial numbers, since that dialog box never shows up. Not sure how we should handle such cases.